### PR TITLE
Add a variable to specifically limit the memory usage of the load part in the insert operation

### PR DESCRIPTION
**Pull request by morningman**
The insert operation is divided into two parts, query and import. Currently, only one variable of `exec_mem_limit` is used to limit the memory of the insert operation. Currently, both load and query has same mem limit, equals to `exec_mem_limit`. But in many cases, queries require more memory and load requires less memory.

Therefore, a `load_mem_limit` variable is added to specify the memory limit of the load part in the insert operation.

Also, revert the CL that setting global variables should not make effect on current session's variables.

ISSUE: #2304 

* Fix bug that variable `enable_insert_strict` does not forward to master FE. #1542 
* Change default strict_mode to false of all kinds of load operation.

### CODE REVIEW

File: be/src/runtime/runtime_state.cpp [hunk 1]
```diff
@@ -531,5 +531,14 @@ Status RuntimeState::get_codegen(LlvmCodeGen** codegen) {
 Status RuntimeState::StartSpilling(MemTracker* mem_tracker) {
     return Status::InternalError("Mem limit exceeded.");
 }
+
+int64_t RuntimeState::get_load_mem_limit() {
+    if (_query_options.__isset.load_mem_limit && _query_options.load_mem_limit > 0) {
```
Comment (imay) [inline be/src/runtime/runtime_state.cpp]: I'm afriad that I don't find that you set this value

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/runtime/runtime_state.cpp (hunk 1 above)
Comment (morningman) [reply be/src/runtime/runtime_state.cpp]: In constructor of Coordinator in FE.
`this.queryOptions = context.getSessionVariable().toThrift();`

Comment (imay) [reply be/src/runtime/runtime_state.cpp]: Oh, I see. However I think it's not a good idea to get all queryOptions from session variables. For example, if we intent to support setting this value in load statement other than through setting variables, then it is not available to get right limit from session variables.
And I think it's better to support this option in stream load.

Comment (morningman) [reply be/src/runtime/runtime_state.cpp]: For broker and stream load, user can set mem limit by `exec_mem_limit` property wrote in load stmt or http request. And I will use `exec_mem_limit` to overwrite `load_mem_limit` in `queryOptions` in query plan.

That is to say, for broker and stream load, user no need to care about this `load_mem_limit`,
and this new variables will make no effect on these kind of load.

File: fe/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -510,6 +431,14 @@ public void write(DataOutput out) throws IOException {
...
         out.writeInt(parallelExecInstanceNum);
         out.writeInt(exchangeInstanceParallel);
+
+        out.writeLong(loadMemLimit);
+        // false means there are no more variables. It you need to add more variables to persist,
+        // change this to true, and add:
+        //      out.writeXXX(new_var_name);
+        //      out.writeBoolean(false);
+        // By doing this, we no longer need to change the FE meta version when adding new variables to persist.
+        out.writeBoolean(false);
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/qe/SessionVariable.java]: I don't this is a elegant way. If we add 100 new variables, the read code will look like ugly.

I think we could serialize and deserialize the whole class by Json.

Comment (morningman) [reply fe/src/main/java/org/apache/doris/qe/SessionVariable.java]: I change it to json.

File: fe/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 3]
```diff
@@ -474,88 +397,140 @@ public TQueryOptions toThrift() {
...
-        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_38) {
-            batchSize = in.readInt();
-            disableStreamPreaggregations = in.readBoolean();
-            parallelExecInstanceNum = in.readInt();
-        }
-        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_62) {
-            exchangeInstanceParallel = in.readInt();
+    }
+
+    private void readFromJson(DataInput in) throws IOException {
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/qe/SessionVariable.java]: I think some json library has already implement this things, like Google Gson.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**